update guide with metadata_hostname change (#193)
hashicorp/terraform-provider-azurestack: website/docs/guides/1.0-upgrade-guide.html.markdown

Version 1.0 of the azurestack Provider is a major release and as such includes some larger-scale changes which are outlined in this document.

-> **NOTE:** Version 1.0 of the azurestack Provider supports Terraform 0.12 and later - Terraform 0.10 & 0.11 are no longer supported with this release.

### Pinning your Provider Version

We recommend pinning the version of each Provider you use in Terraform - you can do this using the `version` attribute in the `provider` block, either to a specific version of the azurestack Provider, like so:

```hcl
provider "azurestack" {
  version = "=1.0.0"
}
```

> .. or to any 0.x release:

Reviewer: ```suggestion
.. or to any 1.x release:
```

At a high level, version 1.0 to includes the following changes:

> * [The `enpoint` property was deprecated in favour of the `metadata_hostname` property](#the-endpoint-property-has-been-replaced-with-metadata_hostname).

Reviewer: ```suggestion
* [The `arm_endpoint` property was deprecated in favour of the `metadata_host` property](#the-arm_endpoint-property-has-been-replaced-with-metadata_host).
```

* [Changes when importing existing resources](#changes-when-importing-existing-resources) - when bringing Resources provisioned outside of Terraform under Terraform's control, these resources will now be required to be imported.
* [Custom Timeouts will be available on Resources](#custom-timeouts-for-resources) - this will allow you to specify a custom timeout for provisioning the resource in your Terraform Configuration [using the `timeouts` block](https://www.terraform.io/docs/configuration/resources.html#operation-timeouts).
* [New resources for Virtual Machines and Virtual Machine Scale Sets](#new-resources-for-virtual-machines-and-virtual-machine-scale-sets).
* [Removing Fields, Data Sources and Resources which have been deprecated](#removal-of-deprecated-fields-data-sources-and-resources).

Each of these topics is covered in more detail below.

> ## The `endpoint` property has been replaced with `metadata_hostname`

Reviewer: ```suggestion
## The `arm_endpoint` property has been replaced with `metadata_host`
```

> The be more descriptive and better align with the AzureRM provider and the provider block property `arm_endpoint` has been deprecated in favour of the `metadata_hostname` property which expects a FQDN hostname instead of a URI. For instance if you were setting it in the provider block previously:

Reviewer: ```suggestion
The be more descriptive and better align with the AzureRM provider and the provider block property `arm_endpoint` has been deprecated in favour of the `metadata_host` property which expects a FQDN hostname instead of a URI. For instance if you were setting it in the provider block previously:
```

```hcl
provider "azurestack" {
  version = "~> 1.x"
  
  arm_endpoint = "https://management.region.mystack.ca"
}
```

It woud now have to be:

```hcl
provider "azurestack" {
  version = "~> 1.x"
  

>   metadata_endpoint = "management.region.mystack.ca"

Reviewer: ```suggestion
  metadata_host = "management.region.mystack.ca"
```

}
```

> Alternatively you if it was being set via environment variables:

Reviewer: ```suggestion
Alternatively, if it was being set via environment variables:
```

```azure

> export ARM_ARM_ENFPOINT="management.region.mystack.ca"

Reviewer: ```suggestion
export ARM_ARM_ENDPOINT="management.region.mystack.ca"
```
